File: PA2_Client/src/edu/ucdenver/tournament/Team.java
package edu.ucdenver.tournament;

import java.util.ArrayList;
import java.util.List;

public class Team {
    private String name;

    private Country country;
    private List<Player> squad;

    public Team(String name, Country country){
        this.name = name;
        this.country = country;
        this.squad = new ArrayList<>();
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Country getCountry(){
        return this.country;
    }

    public List<Player> getSquad(){
        return this.squad;
    }

    public void addPlayer(String name, int age, double height, double weight) throws IllegalArgumentException{
        Player player = new Player(name, age, height, weight);
        if(squad.size() > 35){
            throw new IllegalArgumentException("Player '" + name +
                    "' cannot be added because the max of 35 players has been reached");
        }
        this.squad.add(player);
    }

    @Override
    public String toString(){
        return String.format("\nTeam Name: %s \nTeam Country: %s \n", this.name, this.country);
    }


}

File: PA2_Server/src/edu/ucdenver/Tournament/Team.py
from typing import List

from edu.ucdenver.Tournament import Country
from edu.ucdenver.Tournament.Player import Player


class Team:
    """keeps track of the name, __country, and players on a team"""

    def __init__(self, name: str, country: Country):
        """initializing team object"""
        self.__name = name
        self.__country = country
        self.__squad: List[Player] = []

    @property
    def name(self):
        """returning the name of the team"""
        return self.__name

    @name.setter
    def name(self, n: str):
        self.__name = n

    @property
    def country(self):
        """returning the teams country"""
        return self.__country

    @country.setter
    def country(self, c: Country):
        self.__country = c

    @property
    def squad(self):
        """returning the list of players on the team"""
        return self.__squad

    def add_player(self, name: str, age: int, height: float, weight: float):
        """adding a player to the team"""
        new_player = Player(name, age, height, weight)
        self.__squad.append(new_player)











